### PR TITLE
colima: patch `sw_vers` on darwin

### PR DESCRIPTION
**Pull request by tricktron**
###### Description of changes

- Colima uses `sw_vers` [in util/util.go](https://github.com/abiosoft/colima/blob/feef4176f56a7dea487d43689317a9d7fe9de27e/util/util.go#L141)  to determine the MacOS version and conditionally adds [default start start arguments](https://github.com/abiosoft/colima/blob/feef4176f56a7dea487d43689317a9d7fe9de27e/cmd/start.go#L153). This is also used to check for the new rosetta emulation which is only available on macOS >= 13. We therefore better ~~replace `sw_vers` with the binary from `${darwin.DarwinTools}/bin/sw_vers`~~ add `${darwin.DarwinTools}` to `nativeBuildInputs` so that this logic does not break.
- Build and test all packages instead of only `cmd/colima`. 

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/virtualization/colima/default.nix [hunk 1]
```diff
@@ -42,7 +43,10 @@ buildGoModule rec {
     -X github.com/abiosoft/colima/config.revision=$(cat .git-revision)"
   '';
 
-  subPackages = [ "cmd/colima" ];
+  postPatch = lib.optionalString stdenv.isDarwin ''
+    substituteInPlace util/util.go \
+      --replace 'sw_vers' "${darwin.DarwinTools}/bin/sw_vers"
+  '';
```
Comment (wegank) [inline pkgs/applications/virtualization/colima/default.nix:49]: Could we just put `darwin.DarwinTools` in `nativeBuildInputs` instead of this?

Comment (tricktron) [reply pkgs/applications/virtualization/colima/default.nix]: Yes, indeed. Thank you for the tip.

File: pkgs/applications/virtualization/colima/default.nix [hunk 2]
```diff
@@ -31,7 +32,7 @@ buildGoModule rec {
     '';
   };
 
-  nativeBuildInputs = [ installShellFiles makeWrapper ];
+  nativeBuildInputs = [ installShellFiles makeWrapper darwin.DarwinTools ];
```
Comment (wegank) [inline pkgs/applications/virtualization/colima/default.nix:35]:
```suggestion
  nativeBuildInputs = [ installShellFiles makeWrapper ]
    ++ lib.optionals stdenv.isDarwin [ darwin.DarwinTools ];
```

Comment (tricktron) [reply pkgs/applications/virtualization/colima/default.nix]: Yeah, I realised it just after pushing and corrected it now.

**Review by aaschmid — APPROVED** [review]
LGTM

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 219807` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>colima</li>
  </ul>
</details>

**Pull request merged**